examples/bilingual_document.ipynb

```diff
@@ -0,0 +1,372 @@
+      "text/plain": [
+       "<Figure size 1000x800 with 2 Axes>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "eng_spa_df = pd.DataFrame(eng_spa_cos_sim, index=[f\"Eng {i}\" for i in range(len(eng_sentences))],\n",
+    "                          columns=[f\"Spa {i}\" for i in range(len(spa_sentences))])\n",
+    "\n",
+    "eng_deu_df = pd.DataFrame(eng_deu_cos_sim, index=[f\"Eng {i}\" for i in range(len(eng_sentences))],\n",
+    "                          columns=[f\"Deu {i}\" for i in range(len(deu_sentences))])\n",
+    "\n",
+    "spa_deu_df = pd.DataFrame(spa_deu_cos_sim, index=[f\"Spa {i}\" for i in range(len(spa_sentences))],\n",
+    "                          columns=[f\"Deu {i}\" for i in range(len(deu_sentences))])\n",
+    "\n",
+    "def plot_heatmap(similarity_df, title):\n",
+    "    plt.figure(figsize=(10, 8))\n",
+    "    sns.heatmap(similarity_df, annot=False, cmap=\"YlGnBu\", cbar=True)\n",
+    "    plt.title(title)\n",
+    "    plt.show()\n",
+    "\n",
+    "plot_heatmap(eng_spa_df, \"English-Spanish Cosine Similarity\")\n",
+    "plot_heatmap(eng_deu_df, \"English-German Cosine Similarity\")\n",
+    "plot_heatmap(spa_deu_df, \"Spanish-German Cosine Similarity\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Finding aligned sentence pairs\n",
+    "\n",
+    "We may find which of the sentences correspond to which in the translation by determining the cosine distance and considering equivalent in meaning those sentences that are the closest together in the embedding space.\n",
+    "\n",
+    "`eng_spa_alignment` contains the actual alignment (or mapping) between the indices of (English and Spanish). The 'cosine' distance is used to measure how similar or dissimilar the embeddings are at each point.\n",
+    "\n",
+    "`eng_spa_distance` is the total minimal distance between the two sequences of embeddings the two sequences (English and Spanish embeddings), i.e., which English sentence aligns with which Spanish sentence based on the dynamic time warping (DTW) algorithm.\n",
+    "\n",
+    "#### FastDTW\n",
```

Comment:
The notebook is looking really nice now!
I would keep everything you have implemented as is. 
However, I suggest you add a new section "Monotonic alignment" in the end, where you do actually use DTW.

The goal of this section would be to produce a nice-looking "bilingual book" (a single language pair, like English-Spanish, would be enough) with the following properties:
1. It includes every single sentence both from the English side and the Spanish side (unlike the current threshold-based alignment which e.g. omits the English sentences 14-22).
2. Each sentence is included exactly once (unlike the current alignment, which e.g. prints English Sentence 41 twice).
3. It is monotonic (unlike the current alignment, which e.g. matches English Sentence 41 with Spanish Sentence 12). 
4. It is displayed as a table with well-separated columns, which would make it more readable than the current outputs. 

The combination of properties (1) and (2) implies that if you aren't lucky enough to have exactly 1:1 alignments (which is the case), you'll have to display some sentences as aligned either to nothing, or to multiple sentences.

To obtain the alignment, I suggest you use fastdw, as you have intended originally.
To visualize, there are two options:
1. Implement the table as `pandas DataFrame` and display it. It is easy, but the only disadvantage is that if you have one-to-many alignments, pandas is unable to display it as multiple cells. So you'll have to split each 1-to-many alignment into one 1-to-1 alignment and one or more 1-to-0 alignments. Which is fine.
2. Write your own script for generating an HTML table (which can be visualized with `IPython.display`) where you can assign a `rowspan` property to visualize 1-to-many alignments with "merged" table cells. 
You can choose whichever you like more.